### PR TITLE
ocamlPackages.cstruct: 6.0.1 -> 6.1.0

### PR DESCRIPTION
**Pull request by ulrikstrid**
###### Description of changes

Updates cstruct and reverse dependencies.

This is breaking out parts from https://github.com/NixOS/nixpkgs/pull/165661

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/mirage-block-ramdisk/default.nix [hunk 1]
```diff
@@ -10,7 +10,14 @@ buildDunePackage rec {
...
 
-  minimalOCamlVersion = "4.06";
+  # Make tests compatible with alcotest 1.4.0
+  postPatch = ''
+    substituteInPlace test/tests.ml --replace 'Fmt.kstrf Alcotest.fail' 'Fmt.kstrf (fun s -> Alcotest.fail s)'
+
+    substituteInPlace src/ramdisk.ml --replace 'Cstruct.len' 'Cstruct.length'
+  '';
+
+  minimumOCamlVersion = "4.06";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/mirage-block-ramdisk/default.nix:20]:
```suggestion
  minimalOCamlVersion = "4.06";
```

File: pkgs/development/ocaml-modules/cstruct/default.nix [hunk 2]
```diff
@@ -1,23 +1,22 @@
...
 
 buildDunePackage rec {
   pname = "cstruct";
-  version = "6.0.1";
+  version = "6.1.0";
 
   useDune2 = true;
 
-  minimumOCamlVersion = "4.03";
+  minimumOCamlVersion = "4.08";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/cstruct/default.nix:9]:
```suggestion
  minimalOCamlVersion = "4.08";
```

File: pkgs/development/ocaml-modules/dns/default.nix [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
 }:
 
 buildDunePackage rec {
   pname = "dns";
-  version = "5.0.1";
+  version = "6.2.2";
 
   useDune2 = true;
-  minimumOCamlVersion = "4.07";
+  minimumOCamlVersion = "4.08";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/dns/default.nix:11]:
```suggestion
  minimalOCamlVersion = "4.08";
```

File: pkgs/development/ocaml-modules/git/default.nix [hunk 4]
```diff
@@ -8,14 +8,13 @@
 
 buildDunePackage rec {
   pname = "git";
-  version = "3.5.0";
+  version = "3.9.1";
 
   minimumOCamlVersion = "4.08";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/git/default.nix:13]:
```suggestion
  minimalOCamlVersion = "4.08";
```

File: pkgs/development/ocaml-modules/happy-eyeballs/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+buildDunePackage rec {
+  pname = "happy-eyeballs";
+  version = "0.3.0";
+
+  src = fetchurl {
+    url = "https://github.com/roburio/happy-eyeballs/releases/download/v${version}/happy-eyeballs-${version}.tbz";
+    sha256 = "17mnid1gvq1ml1zmqzn0m6jmrqw4kqdrjqrdsrphl5kxxyhs03m6";
+  };
+  minimumOCamlVersion = "4.08";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/happy-eyeballs/default.nix:13]:
```suggestion
  minimalOCamlVersion = "4.08";
```

File: pkgs/development/ocaml-modules/happy-eyeballs/lwt.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+, happy-eyeballs, cmdliner, duration, dns-client
+, domain-name, ipaddr, fmt, logs, lwt, mtime
+}:
+
+buildDunePackage {
+  pname = "happy-eyeballs-lwt";
+
+  inherit (happy-eyeballs) src version;
+
+  minimumOCamlVersion = "4.08";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/happy-eyeballs/lwt.nix:11]:
```suggestion
  minimalOCamlVersion = "4.08";
```

File: pkgs/development/ocaml-modules/happy-eyeballs/mirage.nix [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+, mirage-time
+, tcpip
+}:
+
+buildDunePackage {
+  pname = "happy-eyeballs-mirage";
+
+  inherit (happy-eyeballs) src version;
+
+  minimumOCamlVersion = "4.08";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/happy-eyeballs/mirage.nix:21]:
```suggestion
  minimalOCamlVersion = "4.08";
```

**Comment by vbgl** [comment]
Can you please move the update of the `dns` libraries to a separate PR for easier review & testing? Thanks.
Suggestions: version 6.3.0 is out; there is no need to set `useDune2`.

**Comment by ulrikstrid** [comment]
@vbgl with all the recent PRs that has landed I've cleaned this PR up to only cstruct, do you have a good idea how I can test this to make sure I didn't break a bunch of packages?
Would be nice if we could find reverse dependencies somehow.

**Comment by vbgl** [comment]
This break `ocamlPackages.git`. I think that cstruct, git, carton, decompress, bigstringaf must be updated simultaneously.

**Pull request closed without merging**